### PR TITLE
DTSBPS-451: Adding processor client

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSBPS-451

### Change description ###
Adding processor client

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/BulkScanProcessorApiProxyTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        String s2sToken = randomUUID().toString();
+        stubFor(put(urlEqualTo("/payment/status"))
+            .withHeader("ServiceAuthorization", equalTo(s2sToken))
+            .withRequestBody(equalToJson(requestJson))
+            .willReturn(aResponse().withStatus(200)));
+
+        proxy.updateStatus(s2sToken, request);
+
+        verify(1, putRequestedFor(urlEqualTo(
+            "/payment/status")).withRequestBody(equalToJson(requestJson)));
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/BulkScanProcessorApiProxyTest.java:50]: formatting

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            .build();
+
+        assertThat(decode(response))
+            .isInstanceOf(HttpServerErrorException.class)
+            .hasMessage(INTERNAL_SERVER_ERROR.value() + " " + INTERNAL_SERVER_ERROR.getReasonPhrase());
+    }
+
+    @DisplayName("Should fail to parse body and throw RuntimeException instead")
+    @Test
+    void should_throw_Failing_Body_Parsing() throws IOException {
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java:69]: naming

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        Request.HttpMethod.GET,
+        "localhost",
+        Collections.emptyMap(),
+        Request.Body.create(new byte[0]),
+        null
+    );
+
+    @DisplayName("Should parse response and return Client specific exception")
+    @Test
+    void should_throw_Client_Exception() {
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java:38]: methods name formats are not consistent.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java (hunk 2 above)
Comment (SomeshRA) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java]: changed to lower case

File: src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java (hunk 3 above)
Comment (SomeshRA) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/processor/CustomFeignErrorDecoderTest.java]: changed to lower case

**Review by SomeshRA — COMMENTED** [review]
Please review

**Review by mdayican — APPROVED** [review]

**Pull request merged**